test/performance/odp_ipsec_ordered.c

```diff
@@ -0,0 +1,1279 @@
+				usage(argv[0]);
+				exit(EXIT_FAILURE);
+			}
+			break;
+		default:
+			break;
+		}
+	}
+
+	optind = 1;             /* reset 'extern optind' from the getopt lib */
+}
+
+int main(int argc, char *argv[])
+{
+	ipsec_args_t *cargs;
+	odp_pool_t pool;
+	odp_queue_param_t qparam;
+	odp_schedule_capability_t schedule_capa;
+	odp_schedule_config_t schedule_config;
+	odp_pool_param_t param;
+	odp_queue_t in_queue = ODP_QUEUE_INVALID;
+	odph_helper_options_t helper_options;
+	odp_instance_t instance;
+	odp_init_t init_param;
+	odp_pool_capability_t capa;
+	odp_ipsec_config_t config;
+	uint32_t max_seg_len;
+	int num_workers;
+	unsigned int i;
+	odp_ipsec_sa_t sa;
+	odp_shm_t shm;
+	ipsec_alg_config_t *algs_config_base;
+	unsigned int num_configs;
+
+	/* Let helper collect its own arguments (e.g. --odph_proc) */
+	argc = odph_parse_options(argc, argv);
+	if (odph_options(&helper_options)) {
+		app_err("Reading ODP helper options failed.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	odp_init_param_init(&init_param);
+	init_param.mem_model = helper_options.mem_model;
+
+	/* Init ODP before calling anything else */
+	if (odp_init_global(&instance, &init_param, NULL)) {
+		app_err("ODP global init failed.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	/* Init this thread */
+	if (odp_init_local(instance, ODP_THREAD_WORKER)) {
+		app_err("ODP local init failed.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	odp_sys_info_print();
+
+	/* Reserve memory for args from shared mem */
+	shm = odp_shm_reserve("shm_args", sizeof(args_t),
+			      ODP_CACHE_LINE_SIZE, 0);
+
+	gbl_args = odp_shm_addr(shm);
+	if (gbl_args == NULL) {
+		ODPH_ERR("Error: shared mem alloc failed.\n");
+		odp_shm_free(shm);
+		exit(EXIT_FAILURE);
+	}
+
+	gbl_args_init(gbl_args);
+	gbl_args->instance = instance;
+	cargs = &gbl_args->cargs;
+
+	/* Parse and store the application arguments */
+	parse_args(argc, argv, cargs);
+	num_workers = cargs->num_workers;
+
+	/* Packet Pool Init */
+	if (odp_pool_capability(&capa)) {
+		app_err("Pool capability request failed.\n");
+		exit(EXIT_FAILURE);
+	}
+	max_seg_len = capa.pkt.max_seg_len;
+
+	for (i = 0; i < sizeof(global_payloads) / sizeof(unsigned int); i++) {
+		if (global_payloads[i] > max_seg_len)
+			break;
+	}
+	global_num_payloads = i;
+	odp_pool_param_init(&param);
+	param.pkt.seg_len = max_seg_len;
+	param.pkt.len      = max_seg_len;
+	param.pkt.num      = POOL_NUM_PKT;
+	param.type         = ODP_POOL_PACKET;
+	pool = odp_pool_create("packet_pool", &param);
+
+	if (pool == ODP_POOL_INVALID) {
+		app_err("packet pool create failed.\n");
+		exit(EXIT_FAILURE);
+	}
+	gbl_args->pkt_pool = pool;
+
+	odp_pool_print(pool);
+
+	/* Scheduler Init */
+	if (odp_schedule_capability(&schedule_capa)) {
+		printf("Error: Schedule capa failed.\n");
+		return -1;
+	}
+	if ((cargs->sched_sync == ODP_SCHED_SYNC_ORDERED) &&
+	    (schedule_capa.max_ordered_locks < 1)) {
+		ODPH_ERR("Error: Ordered locks not available.\n");
+		exit(EXIT_FAILURE);
```

Comment:
This application is not using ordered locks, so there is no point in requiring them here and in the scheduler config.

Reply:
Order lock checks are now removed next version